### ISSUE TITLE
Incorrect behaviour of image modifiers

### ISSUE THREAD

**Issue by mcrakhman** [onevcat/Kingfisher#1609]
Right now it seems `imageModifiers` change the `finalImage` which is cached in memory. This can be a problem when you are using the same image in multiple places. Please correct me if I am wrong. See `ImageCache.swift`. One of the options would be to not cache the final image (or at least add an option to not cache it), or add the same logic to modifiers as we have with processors, so different modifiers would have different keys.
```swift
let finalImage = options.imageModifier?.modify(image) ?? image
// Cache the disk image to memory.
// We are passing `false` to `toDisk`, the memory cache does not change
// callback queue, we can call `completionHandler` without another dispatch.
var cacheOptions = options
cacheOptions.callbackQueue = .untouch
self.store(
    finalImage,
    forKey: key,
    options: cacheOptions,
    toDisk: false)
```

**Comment by onevcat** [comment]
Ummm, I guess you are right on this. Let me think about how can we get it done.

**Comment by mcrakhman** [comment]
Will be so kind and make a new release?

**Comment by onevcat** [comment]
Yes, I will tag it soon after adding more docs to other changes.

**Comment by onevcat** [comment]
Released in 6.1.0.
